### ISSUE TITLE
ProxyError related issues?

### ISSUE THREAD

**Issue by ryanjgillies** [streamlink/streamlink#1244]
### Checklist

- [x] This is a bug report.
- [ ] This is a feature request.
- [ ] This is a plugin (improvement) request.
- [x] I have read the contribution guidelines.

### Description

I'm receiving a number of proxy related errors when trying to use the BBC iPlayer and TVPlayer.com plugins, as follows:

```
streamlink http://www.bbc.co.uk/iplayer/live/bbcone best --config /home/manager/streamlink.cfg -l debug
[cli][info] Found matching plugin bbciplayer for URL http://www.bbc.co.uk/iplayer/live/bbcone
[plugin.bbciplayer][info] A TV License is required to watch BBC iPlayer streams, see the BBC website for more information: https://www.bbc.co.uk/iplayer/help/tvlicence
error: Unable to open URL: https://session.bbc.co.uk/session?ptrt=http%3A%2F%2Fwww.bbc.co.uk%2Fiplayer%2Flive%2Fbbcone (HTTPSConnectionPool(host='session.bbc.co.uk', port=443): Max retries exceeded with url: /session?ptrt=http%3A%2F%2Fwww.bbc.co.uk%2Fiplayer%2Flive%2Fbbcone&context=tvandiplayer&userOrigin=tvandiplayer (Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response',))))
```

```
streamlink https://tvplayer.com/watch/channel4 best --config /home/manager/streamlink.cfg -l debug
[cli][info] Found matching plugin tvplayer for URL https://tvplayer.com/watch/channel4
error: Unable to open URL: https://tvplayer.com/account/login (HTTPSConnectionPool(host='tvplayer.com', port=443): Max retries exceeded with url: /account/login (Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response',))))
```

My config file is as follows:

```
http-proxy=http://user:password@uk97.nordvpn.com
https-proxy=https://user:password@uk97.nordvpn.com
hls-segment-threads=5
hds-segment-threads=5
bbciplayer-username=email@gmail.com
bbciplayer-password=password
tvplayer-email=email@gmail.com
tvplayer-password=password
```

### Expected / Actual behavior

I've never had the BBC iPlayer plugin work for me - previously the message about a tv license prevented me from viewing, although now the ProxyError has appeared after it.

That said, the following command for BBC One has always worked, and still does:
`streamlink "hds://a.files.bbci.co.uk/media/live/manifesto/audio_video/simulcast/hds/uk/pc/llnw/bbc_one_hd.f4m" best --config /home/manager/streamlink.cfg`
So its seemingly not my proxy server that's causing an issue, rather something relating to the remote connection. 

As mentioned in #1228 I had a version issues which I've since resolved, and prior to this error I was receiving "BadStatusLine" errors when using Python2. I've followed @gravyboat 's guidance there on switching to python3, and this is the result.

Has anyone else encountered errors similar to this? What can I do to diagnose the problem?

### Reproduction steps / Explicit stream URLs to test

1. http://www.bbc.co.uk/iplayer/live/bbcone
2. https://tvplayer.com/watch/channel4

### Environment details

Operating system and version: Lubuntu 16.04
Streamlink and Python version: Streamlink 0.8.1 / Python 3.5.2

**Comment by gravyboat** [comment]
Looks to me like you you aren't establishing the connection properly: `Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response',))` I never use the proxy option, maybe try passing it directly on with the command to test and wrap it in quotes. You may also want to use the debug `-l debug` logging to see if that provides more info.

**Comment by ryanjgillies** [comment]
I have now resolved this problem; it was never a streamlink problem in the first place.

To expand on how I figured it out in case anyone has this error, it was indeed a proxy problem as the error description suggested `Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response',))`

After reading that this could possibly be arising [due to HTTPS connections](https://stackoverflow.com/a/33226080) I decided to debug my way line by line through the BBC iPlayer plugin, and sure enough it was the `http.get` request on Line 110 that was throwing the error - the only line (as far as I can tell) that uses a https connection in the plugin.

At first I tried modifying the request to include useragent as the aforementioned stack answer suggested, but that had no effect.

I decided to look again at my proxy settings and realised that for NordVPN I was using a http url for standard proxy and https url for the secure proxy. It was then I realised that I'm using a regular NordVPN proxy (for greater speed), not a secure one. I changed the https-proxy url to a standard http address and sure enough, the problem went away.

So the lesson here is partly that if you receive this error its probably because of your https-proxy settings, but mostly that you shouldn't doubt the streamlink team!

Thanks guys.
